traffic_ops/traffic_ops_golang/crconfig/handler.go

```diff
@@ -78,6 +79,43 @@ func SnapshotGetHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(`{"response":` + snapshot + `}`))
 }
 
+func SnapshotGetMonitoringLegacyHandler(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"cdn"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	snapshot, cdnExists, err := GetSnapshotMonitoring(inf.Tx.Tx, inf.Params["cdn"])
+
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting snapshot: "+err.Error()))
+		return
+	}
+	if !cdnExists {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusNotFound, errors.New("CDN not found"), nil)
+		return
+	}
+	var data tc.TrafficMonitorConfig
+	if err := json.Unmarshal([]byte(snapshot), &data); err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting snapshot: "+err.Error()))
+		return
+	}
+	convertedData := data.ToLegacyConfig()
+
+	type snapshotResponse struct {
+		Response tc.LegacyTrafficMonitorConfig
+	}
+	bytes, err := json.Marshal(snapshotResponse{Response: convertedData})
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting snapshot: "+err.Error()))
+		return
+	}
+	w.Header().Set(rfc.ContentType, rfc.ApplicationJSON)
+	w.Write(bytes)
```

Comment:
`github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api.WriteResp` handles all of this for you, and it makes the output legal contents for a text file (terminating newline).